plugins/communicators/winrm/shell.rb

```diff
@@ -114,30 +112,63 @@ def execute_shell_with_retry(command, shell, &block)
         end
       end
 
-      def raise_winrm_exception(winrm_exception, shell, command)
-        # If the error is a 401, we can return a more specific error message
-        if winrm_exception.message.include?("401")
-          raise Errors::AuthError,
-            user: @username,
-            password: @password,
-            endpoint: endpoint,
-            message: winrm_exception.message
-        end
+      def raise_winrm_exception(exception, shell = nil, command = nil)
+        case exception
+        when WinRM::WinRMAuthorizationError
+          raise Errors::AuthenticationFailed,
+              user: @config.username,
+              password: @config.password,
+              endpoint: endpoint,
+              message: exception.message
+        when WinRM::WinRMHTTPTransportError
+          case exception.status_code
+          # If the error is a 401, we can return a more specific error message
+          when 401
+            raise Errors::AuthenticationFailed,
```

Comment:
This shouldn't be needed with WinRM 1.3, it should always raise a `WinRMAuthorizationError`